### DISCUSSION TITLE
[Android] How can I specify the InfoWindow click event?

### DISCUSSION THREAD

**Discussion by davefxy** [VladislavAntonyuk/MauiSamples#124] [Q&A] [answered]
[Android] How can I add the capture of the InfoWindow click event in the MauiMaps demo?

**Comment by davefxy** [comment, marked as answer]
With net 7.0, when using an ItemSource with the Map control, you can add the following to the Pin DataTemplate:
`       InfoWindowClicked="Pin_InfoWindowClicked".`
When adding pins individually to the Map control, you can set the InfoWindowClicked event like this:
  `      mapPin.InfoWindowClicked += Pin_InfoWindowClicked;
        map.Pins.Add(mapPIn)`
How can I accomplish this assignment within your demo app : MauiMap ? I tried the code listed above in the MauiMap demo on an Android,  and it doesnt work.

**Reply by VladislavAntonyuk** [reply]
Updated sample, click on Pin1 displays toast

**Reply by davefxy** [reply]
Downloaded, compiled and ran it on the Android.
Both images appear on the map. But only the 2nd pin added showed the toast msg when the info window was tapped.
Here is what happened (the order of calls to the handler)
1. CustonMapHandler then MapCallbackHandler
2. ConnectHandler
3. MapPins/AddPins (Map is null)
4. OnMapReady
5. MapPins/AddPins (Map is NOT null

It appears OnMapReady needs to occur in order for Map to have a value.

**Reply by VladislavAntonyuk** [reply]
Yes, I add subscription only to the first pin. You can find it in MainPage.xaml.cs

**Reply by VladislavAntonyuk** [reply]
Map is not null after OnMapReady executes. It is expected

**Reply by davefxy** [reply]
Just realized that was what happened. I was multiplexing. Sorry.

**Reply by VladislavAntonyuk** [reply]
I am glad to help

**Reply by davefxy** [reply]
Is it possible to change your map handler to support itemsource with CustomPins? And Map.ItemTemplate. Perhaps this can wait until Map control is improved to support pin images (imagesource).

**Reply by VladislavAntonyuk** [reply]
I didn't get your question. You can use any pin in Pins property. I don't see any reason to create a separate property to store pins. As for me it is more confusing. The handler only updates the icon for the custom pin. Feel free to edit it for your purposes
